File: src/main/java/diff/patch/accessors/SimpleGetterSetter.java
package diff.patch.accessors;

import diff.patch.Getter;
import diff.patch.Setter;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class SimpleGetterSetter<Type> implements Getter<Type>, Setter<Type> {
    Type obj;

    public SimpleGetterSetter(Type obj) {
        this.obj = obj;
    }

    @Override
    public Type get() {
        return obj;
    }

    @Override
    public void set(Type value) {
        obj = value;
    }
    
}

File: src/main/java/diff/patch/accessors/SimpleGetterSetter.kt
package diff.patch.accessors

import diff.patch.Getter
import diff.patch.Setter

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class SimpleGetterSetter<Type: Any>(internal var obj: Type?) : Getter<Type>, Setter<Type> {

    override fun get(): Type? {
        return obj
    }

    override fun set(value: Type) {
        obj = value
    }

}
